shell/common/run_configuration.h

```diff
@@ -18,38 +18,162 @@
       fml::RefPtr<fml::TaskRunner> io_worker = nullptr);
 
+  //----------------------------------------------------------------------------
+  /// @brief      Creates an run configuration with only an isolate
+  ///             configuration. There is no asset manager and default
+  ///             entrypoint and root library are used ("main" in root library).
+  ///
+  /// @param[in]  configuration  The configuration
+  ///
   RunConfiguration(std::unique_ptr<IsolateConfiguration> configuration);
 
+  //----------------------------------------------------------------------------
+  /// @brief      Creates a run configuration with the spcified isolate
+  ///             configuration and asset manager. The default entrypoint and
+  ///             root library are used ("main" in root library).
+  ///
+  /// @param[in]  configuration  The configuration
+  /// @param[in]  asset_manager  The asset manager
+  ///
   RunConfiguration(std::unique_ptr<IsolateConfiguration> configuration,
                    std::shared_ptr<AssetManager> asset_manager);
 
-  RunConfiguration(RunConfiguration&&);
-
+  //----------------------------------------------------------------------------
+  /// @brief      Run configurations cannot be copied because it may not always
+  ///             be possible to copy the underlying isolate snapshots. If
+  ///             multiple run configurations share the same underlying
+  ///             snapshots, creating a configuration from isolate snapshots
+  ///             sharing the same underlying buffers is recommended.
+  ///
+  /// @param      config  The run configuration to move.
+  ///
+  RunConfiguration(RunConfiguration&& config);
+
+  //----------------------------------------------------------------------------
+  /// @brief      There are no threading restrictions on the destruction of the
+  ///             run configuraion.
+  ///
   ~RunConfiguration();
 
+  //----------------------------------------------------------------------------
+  /// @brief      A valid run configuration only guarantees that the engine
+  ///             should be able to find the assets and the isolate snapshots
+  ///             when it attempts to launch the root isolate. The validity of
+  ///             the snapshot cannot be determined yet. That determination can
+  ///             only be made when the configuration is used to run the root
+  ///             isolate in the engine. However, the engine will always reject
+  ///             an invalid run configration.
+  ///
+  /// @attention  A valid run configuration does no mean that the root isolate
+  ///             will always be launched. It only indicates that the various
+  ///             snapshots are isolate snapshots and asset managers are present
+  ///             and accounted for. The validity of the snapshots will only be
+  ///             checked when the engine attempts to launch the isolate.
+  ///
+  /// @return     Returns if the snapshots is valid.
```

Comment:
pedantic: Returns whether

I initially read that as "hangs if it's not" :P

Reply:
Fixed. Here and later.